akka-actor-tests/src/test/scala/akka/actor/ExtensionSpec.scala

```diff
@@ -73,10 +74,13 @@ class ExtensionSpec extends WordSpec with Matchers {
     "handle extensions that fail to initialize" in {
       val system = ActorSystem("extensions")
 
+      // First attempt, an actor is created and after that
+      // an exception is thrown:
       intercept[FailingTestExtension.TestException] {
         FailingTestExtension(system)
       }
-      // same exception should be reported next time
+
+      // Second attempt, we expect to see the same exception:
```

Comment:
I'm surprised/confused by this behavior. 

Why are we expecting `FailingTestExtension.TestException` when trying to initialize an actor using the same name? Do we cache the exception on the first run?

Reply:
Yes, we cache the registration failure. Clarified that a bit.

Reply:
Maybe the test description should be more explicit then.